src/GraphQL/Execution/Nodes/ObjectExecutionNode.cs

```diff
@@ -42,28 +42,46 @@ public IObjectGraphType GetObjectGraphType(ISchema schema)
         /// <summary>
         /// Returns a representation of the result of this execution node and its children
         /// within a <see cref="Dictionary{TKey, TValue}"/>.
+        /// <see cref="ClearErrorNodes"/> must be called prior to calling this method.
         /// </summary>
         public override object ToValue()
         {
             if (SubFields == null)
                 return null;
 
-            var fields = new ObjectProperty[SubFields.Length];
+            var fields = new Dictionary<string, object>(SubFields.Length);
```

Comment:
Theoretically there could be an empty dictionary, SubFields.Length = 0, but it's OK.